esrally/driver/runner.py

```diff
@@ -1172,16 +1189,18 @@ async def __call__(self, es, params):
         indices = mandatory(params, "indices", self)
         only_if_exists = params.get("only-if-exists", False)
         request_params = params.get("request-params", {})
-
-        for index_name in indices:
-            if not only_if_exists:
-                await es.indices.delete(index=index_name, params=request_params)
-                ops += 1
-            elif only_if_exists and await es.indices.exists(index=index_name):
-                self.logger.info("Index [%s] already exists. Deleting it.", index_name)
-                await es.indices.delete(index=index_name, params=request_params)
-                ops += 1
-
+        prior_destructive_setting = await set_destructive_requires_name(es, False, request_params)
```

Comment:
it occurred to me to do this conditionally but I thought this would be less fragile